rrmngmnt/foreman_host_wrapper.py

```diff
@@ -0,0 +1,186 @@
+
+
+class ForemanHost(Host):
+    """
+    Foreman wrapper class
+    """
+
+    def __init__(
+        self, host_ip, foreman_url, foreman_user,
+        foreman_password, api_version=2
+    ):
+        """
+        Initialize ForemanApi class
+
+        :param host_ip: host ip
+        :type host_ip: str
+        :param foreman_url: foreman url
+        :type foreman_url: str
+        :param foreman_user: foreman user
+        :type foreman_user: str
+        :param foreman_password: foreman password
+        :type foreman_password: str
+        :param api_version: foreman api version
+        :type api_version: int
+        """
+        super(ForemanHost, self).__init__(host_ip)
+        self.foreman_api = Foreman(
+            url=foreman_url,
+            auth=(
+                foreman_user, foreman_password
+            ),
+            api_version=api_version
+        )
+        self.host_id = self.get_element_id(self.fqdn, HOSTS_COLLECTION)
+
+    def get_element_id(
+        self, element_name, collection_name, search_attr="name"
+    ):
+        """
+        Get element id from foreman collection
+
+        :param element_name: element name
+        :type element_name: str
+        :param collection_name: foreman collection name
+        :type collection_name: str
+        :param search_attr: search by attr
+        :type search_attr: str
+        :return: element id
+        :rtype: str
+        """
+        self.logger.info(
+            "Looking for %s under foreman collection %s",
+            element_name, collection_name
+        )
+        collection = getattr(self.foreman_api, collection_name)
+        response = collection.index(
+            search="%s = \"%s\"" % (search_attr, element_name)
+        )
+        if not response["results"]:
+            self.logger.info(
+                "Element '%s' is not declared under foreman collection %s",
+                element_name, collection_name
+            )
+            return ""
+        element = response["results"].pop()
+        self.logger.info("Found element to use: %s", element["name"])
+        return element["id"]
+
+    def build(self):
+        """
+        Build host
+        """
+        self.update(build=True)
+
+    def add(self, mac_address, **kwargs):
+        """
+        Add new host to foreman
+
+        :param mac_address: host mac address
+        :type mac_address: str
+        :param kwargs: location_id = str
+                       domain_id = str
+                       organization_id = str
+                       medium_id = str
+                       architecture_id = str
+                       operatingsystem_id = str
+                       ptable_id = str
+                       hostgroup_id = str
+                       build = bool
+                       root_pass = str
+                       managed = bool
+        :raise: ForemanException
+        """
+        host_d = {
+            "name": self.fqdn,
+            "mac": mac_address,
```

Comment:
it possible to add to foreman offline hosts, so I can not fetch mac address from host